__tests__/app.test.js

```diff
@@ -0,0 +1,46 @@
+const seed = require("../db/seeds/seed.js");
+const { categoryData, commentData, reviewData, userData } = require("../db/data/test-data/index.js");
+const db = require("../db/connection.js");
+const request = require("supertest");
+const app = require("../index.js");
+
+
+beforeEach(() => seed({ categoryData, commentData, reviewData, userData }));
+
+afterAll(() => db.end());
+
+describe("404 endpoint", () => {
+    test("If invalid url is sent, output 404 status code and a message", () => {
+        return request(app)
+            .get("/api/invalid_url")
+            .expect(404)
+            .then(({ body: { msg } }) => {
+                expect(msg).toBe("Invalid Path");
+            });
+    });
+});
+
+describe("GET Categories", () => {
+    const testCategory = {
+        slug: expect.any(String),
+        description: expect.any(String)
+    }
+    test("Test that /api/categories returns 200", () => {
+        return request(app)
+            .get("/api/categories")
+            .expect(200)
+    });
+    
+    test("Test that /api/categories returns the correct type of object", () => {
+        return request(app)
+            .get("/api/categories")
+            .expect(200)
+            .then(({ body }) => {
+                const { categories } = body;
+                expect(categories).toHaveLength(4);
+                expect(categories).toEqual(
+                    expect.arrayContaining([expect.objectContaining(testCategory)])
+                );
```

Comment:
Id prefer you do this with a forEach. arrayContaining, will check as to whether the thing exists in the array but it only needs to be true the once. We dont know whether that every object looks like this. With a forEach we can check each of the objects match this structure